django_project/frontend/src/pages/Dashboard/MapLibre/LayerType/RasterCog.jsx

```diff
@@ -63,31 +66,31 @@ export default function rasterCogLayer(map, id, data, setData, contextLayerData,
         colors: colors,
       }
 
+      const key = id + JSON.stringify(requestBody);
       let classifications = [];
-      const session = new Session(id, 1000)
-      if (!session.isValid) {
-        return
-      }
-
-      await DjangoRequests.post(
-        `/api/raster/classification`,
-        requestBody
-      ).then(response => {
-        response.data.forEach((threshold, idx) => {
-          if (idx < response.data.length - 1) {
-            classifications.push({
-              bottom: threshold,
-              top: response.data[idx + 1],
-              color: colors[idx]
-            });
-          }
-        });
-      }).catch(error => {
-        throw Error(error.toString())
-      })
 
-      if (!session.isValid) {
-        return
+      // if classfication for the request body exist, use it
+      // otherwise, get it from API
+      if (sessions[key]) {
+        classifications = sessions[key];
+      } else {
+        await DjangoRequests.post(
```

Comment:
Could you make sure it is not being called twice or more as the previous one is not done yet?